app/routers/organizations.py

```diff
@@ -1,15 +1,57 @@
 from fastapi import APIRouter, status, HTTPException
 from database import client
-from models import Organization
+from models import Organization, OrganizationResponse
+import secrets
+import string
+from fastapi.encoders import jsonable_encoder
 
-router = APIRouter()
+router = APIRouter(prefix="/organizations", tags=["Organizations"])
 
 
-@router.get("/check-auth-token/{api_key}", response_model=Organization)
+@router.post("/", response_model=OrganizationResponse)
+async def create_organization(org: Organization):
+    org = jsonable_encoder(org)
+    if org["name"] is not None:
+
+        # create an API key
+        key = "".join(
+            [secrets.choice(string.ascii_letters + string.digits) for _ in range(20)]
+        )
+        number_of_loops = 3
+
+        while number_of_loops > 0:
```

Comment:
Also I don't really think that this extra logic is needed.
it would be very rare for a 20 character alphanumeric string to repeat itself.
and on top of that, we're just creating only one organization as of now :p 

A few things to help clean up things here
- create a separate function/method for creating random strings. Because I can see below that the API key generation code is being repeated. Why repeat the code if you can make it into a function? This function can either be here, or in some other module.
- assuming the function you created is called `generate_random_string()`, you can refactor the code as below
```py
@router.post("/", response_model=OrganizationResponse)
async def create_organization(org: Organization):
    org = jsonable_encoder(org)
   api_key = generate_random_string()
   while client.quiz.organization.find_one({"key": api_key}) is not None:
      api_key = generate_random_string()

   org["key"] = api_key
   new_org = client.quiz.organization.insert_one(org)
   created_org = client.quiz.organization.find_one(
      {"_id": new_org.inserted_id}
   )
   return created_org
```

I don't think anything else is needed.